app/controllers/floors_controller.rb

```diff
@@ -51,7 +51,7 @@ def add_location
       location = update_location(l) if l['id'].present?
       label = find_or_create_label(location, floor_params[:label_attributes])
       icon = save_location_on_icon(location, floor_params[:icon_attributes][:id])
-      locations << { location: location, label: label, icon: icon }
+      icon.present? ? locations << { location: location, label: label, icon: icon } : locations << { location: location, label: label }
```

Comment:
Includes an icon or not depending on if icon is present

Reply:
```
data = { location: location, label: label }
data[:icon] = icon if icon.present?
locations << data
```